src/http/proxy_server.rs

```diff
@@ -11,22 +11,21 @@ use hyper::Response;
 use hyper::Server;
 use hyper::StatusCode;
 
+async fn handle(req: Request<Body>, port: u16) -> Result<Response<Body>, anyhow::Error> {
+    let fpm_url = format!("http://127.0.0.1:{}", port);
+
+    handle_request(fpm_url.clone(), req).await
+}
+
 #[tokio::main]
 pub(crate) async fn start(port: u16) {
     let addr = SocketAddr::from(([127, 0, 0, 1], port));
 
-    let fpm_url = format!("http://127.0.0.1:{}", port);
-
-    let make_service_fn = make_service_fn(move |_socket: &AddrStream| async {
-        let fpm_url = fpm_url.clone();
-        let request_handler = move |req: Request<Body>| async {
-            handle_request(fpm_url.clone(), req)
-        };
-        let service = service_fn(request_handler);
-        Result::<_, Infallible>::Ok(service)
+    let make_svc = make_service_fn(|_conn| async {
```

Comment:
```suggestion
    let make_svc = make_service_fn(|_conn| async move {
```

This will move `port` and make it available in the closure

Reply:
Tried it but it does not work ;)